compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/optimizer/LargeMethodOptimizer.java

```diff
@@ -104,28 +122,1064 @@ public void splitLargeBIRFunctions(BIRPackage birPkg) {
+            }
+            if (isTempOrSyntheticVar(bbIns.lhsOp.variableDcl)) {
+                splitFuncEnv.splitFuncLocalVarList.add(bbIns.lhsOp.variableDcl);
+            }
+        }
+    }
+
+    private TempVarsForArraySplit getTempVarsForArraySplit() {
+        BIRVariableDcl arrayIndexVarDcl = new BIRVariableDcl(null, symbolTable.intType,
+                new Name("%arrIndex"), VarScope.FUNCTION, VarKind.TEMP, null);
+        BIRVariableDcl typeCastVarDcl = new BIRVariableDcl(null, symbolTable.anyOrErrorType,
+                new Name("%typeCast"), VarScope.FUNCTION, VarKind.TEMP, null);
+        return new TempVarsForArraySplit(arrayIndexVarDcl, typeCastVarDcl);
+    }
+
+    private boolean setMapElementGivenOperand(Map<BIROperand, BIRMappingConstructorEntryWithIndex> birOperands,
+                                              List<BIRNonTerminator> newInsList, BIROperand handleArrayOperand,
+                                              TempVarsForArraySplit tempVars,
+                                              Map<BIRAbstractInstruction, SplitPointDetails> insSplitPoints,
+                                              BIRAbstractInstruction currIns, SplitFuncEnv splitFuncEnv,
+                                              Map<BIROperand, NonTerminatorLocation> mapKeyOperandLocations,
+                                              int bbIndex, int insIndex, Map<BIRAbstractInstruction,
+            BIRMappingConstructorEntryWithIndex> globalAndArgVarKeyOrValueRelatedIns) {
+        boolean splitPointsContainCurrIns = insSplitPoints.containsKey(currIns);
+        if (splitPointsContainCurrIns) {
+            SplitPointDetails splitPointDetails = insSplitPoints.get(currIns);
+            List<BIROperand> currOperands = splitPointDetails.arrayElementsBIROperands;
+            for (BIROperand currOperand : currOperands) {
+                BIRMappingConstructorEntryWithIndex mappingConstructorEntryWithIndex = birOperands.get(currOperand);
+                BIRNode.BIRMappingConstructorEntry mappingConstructorEntry = mappingConstructorEntryWithIndex
+                        .mappingConstructorEntry;
+                boolean isKeyValueEntry = mappingConstructorEntry instanceof BIRNode.BIRMappingConstructorKeyValueEntry;
+                if (isKeyValueEntry && insIndex != -1) {
+                    BIROperand keyOperand = ((BIRNode.BIRMappingConstructorKeyValueEntry) mappingConstructorEntry)
+                            .keyOp;
+                    if (mapKeyOperandLocations.containsKey(keyOperand)) {
+                        NonTerminatorLocation nonTerminatorLocation = mapKeyOperandLocations.get(keyOperand);
+                        if (nonTerminatorLocation != null && (nonTerminatorLocation.bbNum > bbIndex ||
+                                (nonTerminatorLocation.bbNum == bbIndex && nonTerminatorLocation.insNum > insIndex))) {
+                            splitPointDetails.splitHere = false;
```

Comment:
Seem this if condition does not get evaluate

Reply:
This will get covered by a valid Ballerina code as shown in https://github.com/ballerina-platform/ballerina-lang/pull/41181/files#diff-fd8491d71ee6c15cb7ba546e20e8e255b9d37ea87e5d612ac87c9de2b5ec1fa8R1049-R1052. I'll try to add a test for this.

Reply:
This if condition evaluates to true if I run the existing `TableWithXMLKeySpecifierTest` test by setting low thresholds as done [here](https://github.com/ballerina-platform/ballerina-lang/pull/41181). I tried to add a test but it is difficult with higher thresholds since the BIR should contain a map value operand reused and at its first occurrence the key operand should not be there before that. It is difficult to write Ballerina code to produce that BIR. Because the line is covered with lower thresholds, will keep this change and will not remove this.